Library/Homebrew/extend/os/linux/extend/ENV/super.rb

```diff
@@ -3,4 +3,11 @@ module Superenv
   def self.bin
     (HOMEBREW_SHIMS_PATH/"linux/super").realpath
   end
+
+  def determine_rpath_paths
+    paths = ["#{HOMEBREW_PREFIX}/lib"]
+    paths += deps.map { |d| d.opt_lib.to_s }
```

Comment:
Should this be restricted to runtime dependencies?

For `shellinabox` I got:

```bash
patchelf --print-rpath ~/.linuxbrew/bin/shellinaboxd 
/home/linuxbrew/.linuxbrew/lib:/home/linuxbrew/.linuxbrew/opt/libtool/lib:/home/linuxbrew/.linuxbrew/opt/zlib/lib:/home/linuxbrew/.linuxbrew/opt/openssl/lib
```

Putting some debugging code in revealed `autoconf` and `automake`, both build deps, were also considered, but they didn't end up in the RPATH presumably because they don't have lib directories.